src/selection-match.ts

```diff
@@ -59,27 +74,35 @@ const matchHighlighter = ViewPlugin.fromClass(class {
     let conf = view.state.facet(highlightConfig)
     let {state} = view, sel = state.selection
     if (sel.ranges.length > 1) return Decoration.none
-    let range = sel.main, query, check = null
+    let range = sel.main, query, check = null, empty = false
     if (range.empty) {
       if (!conf.highlightWordAroundCursor) return Decoration.none
       let word = state.wordAt(range.head)
       if (!word) return Decoration.none
+      empty = true
```

Comment:
Would be simpler to just check `range.empty` later on, instead of defining a variable for it

Reply:
It does look like `if (range.empty)` would have the same effect, good idea.